XDMA/linux-kernel/xdma/xdma_netdev.c

```diff
@@ -243,7 +244,37 @@ void xdma_tx_work(struct work_struct *work) {
                 return;
         }
 
-        shhwtstamps.hwtstamp = ns_to_ktime(alinx_get_tx_timestamp(priv->pdev, metadata->timestamp_id));
+        /*
+         * Read TX timestamp several times because
+         * 1. Reading and writing TX timestamp register are not atomic
+         * 2. The work thread might try to read TX timestamp before the register gets updated
+         */
+        tx_tstamp = alinx_read_tx_timestamp(priv->pdev, metadata->timestamp_id);
+        if (tx_tstamp == priv->last_tx_tstamp && priv->tstamp_retry < TX_TSTAMP_MAX_RETRY) {
+                /*
+                 * Tx timestamp is not updated. Try again.
+                 * Waiting for it to be updated forever is not desirable,
+                 * so limit the number of retries
+                 */
+                priv->tstamp_retry++;
+                if (priv->tstamp_retry == TX_TSTAMP_MAX_RETRY) {
```

Comment:
```suggestion
                if (priv->tstamp_retry >= TX_TSTAMP_MAX_RETRY) {
```

범위 체크는 항상 보수적으로 하는 게 좋습니다.